plugins/modules/import_eks.py

```diff
@@ -245,17 +247,24 @@ def execute_module(module):
     )
 
     # TODO: handle error if eks cluster does not exist or unable to connect
-    eks_kubeconfig = get_eks_kubeconfig(eks_conn, sts_token, eks_cluster_name)
+    try:
+        eks_kubeconfig = get_eks_kubeconfig(eks_conn, sts_token, eks_cluster_name)
+    except:
+        module.fail_json(msg="Error no cluster found for name: {0}".format(eks_cluster_name))
+
     eks_kubeconfig.verify_ssl = False
     eks_kube_client = kubernetes.dynamic.DynamicClient(
         kubernetes.client.api_client.ApiClient(configuration=eks_kubeconfig)
     )
 
     try:
-        managedcluster = import_utils.ensure_managedcluster(hub_client, eks_cluster_name)
-        import_utils.ensure_klusterletaddonconfig(hub_client, eks_cluster_name, addons)
+        managedcluster = import_utils.ensure_managedcluster(hub_client, eks_cluster_name, timeout)
+        import_utils.ensure_klusterletaddonconfig(hub_client, eks_cluster_name, addons, timeout)
         if import_utils.should_import(managedcluster):
-            crds_yaml, import_yamls = import_utils.get_import_yamls(hub_client, eks_cluster_name)
+            if import_utils.is_namespace_exists(eks_kube_client, "open-cluster-management-agent"):
+                module.fail_json(msg="Error open-cluster-management-agent namespace already exists in {0} cluster".format(eks_cluster_name))
```

Comment:
check if klusterlet exist rather than if the namespace exist

Reply:
Done replaced namespace check with klusterlet check